src/pybind/mgr/cephadm/designs/host-maintenance.rst

```diff
@@ -0,0 +1,100 @@
+================
+Host Maintenance
+================
+
+All hosts that support Ceph daemons need to support maintenance activity, whether the host
+is physical or virtual(vm or cloud). This means that the management workflows should provide
+a simple and consistent way to support this operational requirement. This document defines
+the maintenance strategy that will be implemented in cephadm and mgr/cephadm.
+
+
+High Level Design
+=================
+Placing a host into maintenance, adopts the following workflow;
+
+#. confirm that the removal of the host does not impact data availability (the following
+   steps will assume it is safe to proceed)
+
+   * orch host ok-to-stop <host> would be used here
+
+#. if the host has osd daemons, apply noout to the host subtree to prevent data migration
+   from triggering during the planned maintenance slot.
+#. Stop the ceph target (all daemons stop)
```

Comment:
this sounds like a `orch host drain` command?

Reply:
Don't think so. A host drain would be the removal of services, so you can ultimately remove the host from the cluster. What I'm talking about here is supporting a maintenance slot, which would normally be for a couple of hours on a host - with the aim to avoid data movement since the slot is of a 'known'/planned duration.

Make sense?